user/jwt.md

```diff
@@ -0,0 +1,155 @@
+---
+title: JWT addon
+layout: en
+permalink: /user/jwt/
+---
+
+Integration between Travis-CI and third-party services like Sauce Labs relies
+on [encrypted variables](http://docs.travis-ci.com/user/environment-variables/#Encrypted-Variables)
+which works well for trusted branches and committers.
+For security reasons, encrypted variables are not exposed to untrusted pull requests,
+so builds of pull requests do not have access to third party integrations.
+
+
+The JWT addon replaces encrypted variables with a time-limited authentication
+token, which is exposed to pull requests without security consequences.
+
+For this to work the JWT addon needs to be enabled in the `.travis.yml` file,
+and the third-party need to have integrated with the JWT service and allow
+token-based authentication.
+
+<img src="../images/travis_jwt.svg" alt="JWT Travis Flow Diagram">
+
+
+### .travis.yml
+
+Add the encrypted key to the `jwt` section of the `.travis.yml` file.
+This can be done manually or using the `travis encrypt` command
+
+Travis Encrypt:
+
+`travis encrypt --add addons.jwt SAUCE_ACCESS_KEY=your-access-key`
+
+Manually:
+
+```yaml
+addons:
+  jwt:
+     secure: <SAUCE_ACCESS_KEY ENCRYPTED>
+```
+
+This can also support several services:
+
+`travis encrypt --add addons.jwt SAUCE_ACCESS_KEY=your-access-key THIRDPARTY_SHARED_SECRET=another-key`
+
+Manually:
+
+```yaml
+addons:
+  jwt:
+    - secure: <SAUCE_ACCESS_KEY ENCRYPTED>
+    - secure: <THIRDPARTY_SHARED_SECRET ENCRYPTED>
+```
+
+### Use the Encrypted Key
+
+The original variable names are available within the Travis CI build as
+environment variables containing the JWT tokens instead of the original values.
+
+For example, using the previous configuration `SAUCE_ACCESS_KEY` and
+`THIRDPARTY_SHARED_SECRET` will be available as environment variables.
+
+### How secure is this addon?
+
+The JWT token is only valid for 90 minutes. It is signed in a way that lets you securely
+transmit your secret information without worrying that it is leaked.
+
+### Troubleshooting
+
+1. Check if the third-party service is [supported](#List-of-Third-Party-Services-Integrated-with-the-JWT-Addon)
+2. Contact the third-party support and provide them with the encrypted token (echo the key in your test script), and link to the Travis job.
+
+## Third-Party Service Integration
+
+Third-party service needs to implement a new authentication method on the server side so that the JWT token is recognized and verified.
+
+### JWT Libraries
+
+[JWT.io](https://jwt.io) has a complete list of [supported languages](https://jwt.io/#libraries-io) and documentation on how everything works
+
+### Payload
+
+An example payload used to generate the JWT token:
+
+```javascript
+{
+  "iss": "Travis CI, GmbH",
+  "slug": "<SLUG>",
+  "pull-request": "<PR>",
```

Comment:
I think it's actually a number or blank.

I'll be back from the conference this weekend, so I'll update the pr next
week

On Aug 3, 2016 9:50 AM, "Hiro Asari" <notifications@github.com> wrote:

> In user/jwt.md
> <https://github.com/travis-ci/docs-travis-ci-com/pull/674#discussion_r73335205>
> :
>
> > +
> > +Third-party service needs to implement a new authentication method on the server side so that the JWT token is recognized and verified.
> > +
> > +### JWT Libraries
> > +
> > +[JWT.io](https://jwt.io) has a complete list of [supported languages](https://jwt.io/#libraries-io) and documentation on how everything works
> > +
> > +### Payload
> > +
> > +An example payload used to generate the JWT token:
> > +
> > +```javascript
> > +{
> > +  "iss": "Travis CI, GmbH",
> > +  "slug": "<SLUG>",
> > +  "pull-request": "<PR>",
>
> <PR> requires explanation. It is either true or ''.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/travis-ci/docs-travis-ci-com/pull/674/files/bdd4590543bc0ba02dfdf64efb46decb8f0e334d#r73335205>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAGuB1MsHOyzbEfw9oNF0LY3_fgPFoFLks5qcJZZgaJpZM4JQSxS>
> .
>